### PR TITLE
update code to support sklearn v1.2

### PR DESCRIPTION
**Pull request by OkonSamuel**
At the time of opening the PR the latest version of python's scikitlearn is v1.2.1. In line with this new version the following changes have been made.

1. The `normalize` parameter have been removed from the following linear models

- `ARDRegressor`
- `BayesianRidgeRegressor`
- `ElasticNetRegressor`
- `ElasticNetCVRegressor`
- `LassoRegressor`
- `LassoCVRegressor`
- `LinearRegressor`
- `MultiTaskElasticNetRegressor`
- `MultiTaskElasticNetCVRegressor`
- `MultiTaskLassoRegressor`
- `MultiTaskLassoCVRegressor`
- `RidgeCVClassifier`
- `RidgeClassifier`

2. The `normalize` parameter of the following linear models has been set as `false`, but will be removed when python scikitlearn v1.4 is released. 
- `OrthogonalMatchingPursuitRegressor`
- `OrthogonalMatchingPursuitCVRegressor`
-  `LarsRegressor`
- `LarsCVRegressor`
- `LassoLarsCVRegressor`
- `LassoLarsICRegressor`

3. The `base_estimator` parameter in `AdaBoostRegressor`, `AdaboostClassifier`, `BaggingRegressor`, `BaggingClassifier` and `RANSACRegressor` models has been renamed to `estimator`.

4. The `absolute_loss` and `squared_loss` losses has been replaced by the equivalent
`absolute_error` and `squared_error` respectively.

5. Depreciate the `auto` option for `max_features` in `RandomForestRegressor`, `RandomForestClassifier`, `ExtraTreesRegressor` and `ExtraTreesClassifier` models. The new default for `max_features` has been set to
`sqrt`.
6. Replace the `ls` loss with the equivalent `squared_loss` error. Set the default value of `loss` parameter in the `GradientBoostingRegressor` model to `squared_loss`
7. Default value for `criterion` in `RandomForestRegressor`and `ExtrasTreesRegressor` models has been set to `squared_loss`. New losses such as `friedman_mse` and `poisson` has been added to the list of possible options for `criterion`
8. `log_loss` has been added to the list of possible option for the `criterion` parameter in `RandomForestClassifier` and `ExtrasTreesClassifier` models.
9. The `lad` loss in the `GradientBoostingRegressor` model has been replaced by the equivalent `absolute_error` loss
Additionally
- The `mae` loss was removed from the possible options for the `criterion`parameter
- The `mse` was replaced with the equivaled `squared_error` and, 
- The `friedman_mse` was added to the possible options for the `criterion` parameter.
10. Add the `"log_loss"` loss as one of the possible options to the `loss` parameter in the `ProbalisticSGDClassifier`and `SGDClassifier` models.
11. Change the default value of `loss` parameter in `ProbalisticSGDClassifier` model from `"log"` to `"log_loss"`.
12. Change the default value of `algorithm` parameter in `KMeans` model from `"auto"` to `"lloyd"`.
 
see https://scikit-learn.org/stable/whats_new/v1.0.html#changes-1-0 and
https://scikit-learn.org/stable/whats_new/v1.0.html

### CODE REVIEW

**Comment by ablaom** [comment]
@tylerjthomas9 I am pinging you here to make sure we avoid any duplication of effort on this front, as I know you are also working on a fork of this repo.

**Review by ablaom — COMMENTED** [review]
Looks good to me. Let's be sure to:

- [ ] Check CI log for any persistent warnings

once that is passing.

**Comment by OkonSamuel** [comment]
I have opened a PR at https://github.com/cstjean/ScikitLearn.jl/pull/119. This PR should pass once that PR is merged.

**Comment by OkonSamuel** [comment]
> Looks good to me. Let's be sure to:
> 
> * [ ]  Check CI log for any persistent warnings
> 
> once that is passing.

Some warning in the ci can't be fixed at the moment except I remove the `normalize` argument completely. But the thing is that python sklearn won't remove this argument till v1.4 comes out.

**Comment by ablaom** [comment]
The CI log is showing some deprecation warnings. @OkonSamuel Can you please comment on these.

**Comment by ablaom** [comment]
Okay, as per zoom call, we'll not address those dep warnings yet.
